terraform/deployments/cluster-services/argo_workflows.tf

```
@@ -0,0 +1,46 @@
+locals {
+  tag_image_service_account_name = "add-tag-to-image-workflow"
+}
+
+module "tag_image_iam_role" {
+  source  = "terraform-aws-modules/iam/aws//modules/iam-role-for-service-accounts-eks"
+  version = "~> 5.5"
+
+  role_name        = "${local.tag_image_service_account_name}-${data.terraform_remote_state.cluster_infrastructure.outputs.cluster_id}"
+  role_description = "Role for the add-tag-to-image Argo Workflow. Corresponds to ${local.tag_image_service_account_name} k8s ServiceAccount."
+
+  oidc_providers = {
+    main = {
+      provider_arn               = data.terraform_remote_state.cluster_infrastructure.outputs.cluster_oidc_provider_arn
+      namespace_service_accounts = ["${var.apps_namespace}:${local.tag_image_service_account_name}"]
+    }
+  }
+}
+
+data "aws_iam_policy_document" "tag_image" {
+  statement {
+    actions = [
+      "ecr:GetAuthorizationToken",
+    ]
```

optional tiniest of nits: maybe slightly cleaner to write this on a single line like on line 33
```suggestion
    actions = ["ecr:GetAuthorizationToken"]
```